Sprint 4 Selenium Project

bahek091 opened the pull request:
Реализованы автотесты по проекту 4 спринта:
 - полная цепочка создания заказа через кнопку в хедере и в теле страницы
 - проверка соответствия текста вопросам в разделе  FAQ
 - сообщения об ошибках ввода на странице заказа
 - поиск заказа - страница об отсутствии заказа
 - клик на логотипы яндекса и самоката в хедере

Commit:
Added all files for project

pomegranate282922 commented on src/main/java/org/example/pageobject/MainPage.java:
```diff
@@ -0,0 +1,101 @@
+    private By cookieButton = By.xpath(".//button[contains(@class, 'App_CookieButton')]");
+    private By faqAccordionQuestionDiv = By.xpath(".//div[contains(@class, 'accordion__heading')]");
+    private By faqOppenedAnswerP = By.xpath(".//div[contains(@class, 'accordion__panel') and not(@hidden)]/p");
+    private final String FAQ_QUESTION_PATTERN = ".//div[contains(@id, 'accordion__heading') and contains(text(), '%s')]";
```
Не статичные константы именуем camelCase 
Нужно исправить

pomegranate282922 commented on src/main/java/org/example/pageobject/MainPage.java:
```diff
@@ -0,0 +1,101 @@
+    }
+
+
+    private void waitBeforeClick(int timeout, By element) {
```
Это лучше вынести в предок типа BasePage 
Можно улучшить

pomegranate282922 commented on src/main/java/org/example/pageobject/OrderPage.java:
```diff
@@ -0,0 +1,140 @@
+    private By formModalSuccessMessage = By.xpath(".//div[text() = 'Заказ оформлен']");
+
+
+    private final String METRO_PATTERN = ".//*[contains(text(), '%s')]";
```
camelCase

pomegranate282922 commented on src/test/java/org/example/tests/BaseUITest.java:
```diff
@@ -0,0 +1,37 @@
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+public class BaseUITest {
```
Отлично что есть иерархия тестов

pomegranate282922 commented on src/test/java/org/example/tests/CheckFAQTextTest.java:
```diff
@@ -0,0 +1,35 @@
+                {"Сколько это стоит? И как оплатить?", "Сутки — 400 рублей. Оплата курьеру — наличными или картой."},
+                {"Можно ли заказать самокат прямо на сегодня?", "Только начиная с завтрашнего дня. Но скоро станем расторопнее."},
+                {"Я жизу за МКАДом, привезёте?","Да, обязательно. Всем самокатов! И Москве, и Московской области."}
+        };
```
Нужно проверить все 8 вопросов 
Нужно исправить

pomegranate282922 commented on src/test/java/org/example/tests/CheckFAQTextTest.java:
```diff
@@ -0,0 +1,35 @@
+    public void shouldBeCorrectFAQAnswersTest(){
+        MainPage mainPage = openMainPage(driver);
+
+        Assert.assertTrue("Answer differs from origin value.", answerText.equals(mainPage.getFAQAnswerByQuestionText(questionText)));
```
Лучше через assertEquals - сообщение при падении будет более понятным 
Можно улучшить

pomegranate282922 reviewed (commented):
Работа хорошая! Здорово что выполнено доп задание. Применена параметризация.
Замечания:
1. Нейминг ( просмотреть по всему коду)
2. Доделать все вопросы в faq 
3. Остальное по мелочи

Commit:
Выполнил рекомендованные улучшения кода:
-поправил именование переменных
-добавил все вопросы из FAQ
-вынес общий метод в BasePage

Commit:
Выполнил рекомендованные улучшения кода:
-поправил именование переменных
-добавил все вопросы из FAQ
-вынес общий метод в BasePage

Commit:
Merge branch 'dev' of https://github.com/bahek091/Sprint_4 into dev

# Conflicts:
#	src/main/java/org/example/pageobject/MainPage.java

bahek091 commented:
Динара, спасибо большое за отзыв!
Внес правки и улучшения

pomegranate282922 commented on src/test/java/org/example/tests/CheckOrderStatusTest.java:
```diff
@@ -0,0 +1,24 @@
+
+
+    @Test
+    public void OrderStatusNotExistTest() {
```
camelCase

pomegranate282922 reviewed (approved):
Все ок исправил ( ну почти :)
Зачет!
